src/components/AccordionButton/AccordionButton.tsx

```diff
@@ -0,0 +1,95 @@
+import { Disclosure } from '@headlessui/react';
+import clsx from 'clsx';
+import React, { useContext } from 'react';
+import styles from './AccordionButton.module.css';
+import { ENTER_KEYCODE, SPACEBAR_KEYCODE } from '../../util/keycodes';
+import { AccordionContext } from '../Accordion';
+import Button from '../Button';
+import type { HeadingElement } from '../Heading';
+import Heading from '../Heading';
+import Icon from '../Icon';
+
+export type Props = {
+  /**
+   * Child node(s) that can be nested inside component.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classnames passed in for styling
+   */
+  className?: string;
+  /**
+   * Used to specify which heading element should be rendered for the title.
+   */
+  headingAs: HeadingElement;
+  /**
+   * Callback called when accordion is closed.
+   */
+  onClose?: () => void;
+};
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * The button subcomponent that houses the heading text and open indicator icon for the EDS Accordion component.
+ */
+export const AccordionButton = ({
+  children,
+  className,
+  headingAs,
+  onClose,
+  ...other
+}: Props) => {
+  const { variant } = useContext(AccordionContext);
+
+  const componentClassName = clsx(
+    styles['accordion-button'],
+    variant === 'compact' && styles['accordion-button--compact'],
+    className,
+  );
+
+  const headingClassName = clsx(
+    styles['accordion-button__heading'],
+    variant === 'compact' && styles['accordion-button__heading--compact'],
+  );
+
+  return (
+    <Disclosure.Button as={React.Fragment}>
+      {({ open }) => (
+        <Button
+          className={componentClassName}
+          fullWidth
+          onClick={() => {
+            if (open && onClose) {
+              onClose();
+            }
+          }}
+          onKeyDown={(e) => {
+            if (e.key === SPACEBAR_KEYCODE || e.key === ENTER_KEYCODE) {
+              if (open && onClose) {
+                onClose();
+              }
+            }
+          }}
+          status="neutral"
+          variant="icon"
+          {...other}
+        >
+          <Heading className={headingClassName} size={headingAs}>
+            {children}
```

Comment:
Removed the `Accordion.Title` subcomponent for use in `Accordion.Button` directly. The styling is hard to override anyways to begin with due to double class selectors and this can be selected in LP via `> h2` or wrap their title in a styled element if they really disagree with the EDS styling